tests/sentry/ingest/test_transaction_clusterer.py

```diff
@@ -84,30 +87,65 @@ def test_collection():
 
 def test_clear_redis():
     project = Project(id=101, name="p1", organization=Organization(pk=66))
-    _store_transaction_name(project, "foo")
+    _record_sample(ClustererNamespace.TRANSACTIONS, project, "foo")
     assert set(get_transaction_names(project)) == {"foo"}
-    clear_transaction_names(project)
+    clear_samples(ClustererNamespace.TRANSACTIONS, project)
     assert set(get_transaction_names(project)) == set()
 
-    # Deleting for a none-existing project does not crash:
+    # Deleting for a non-existing project does not crash:
     project2 = Project(id=666, name="project2", organization=Organization(pk=66))
-    clear_transaction_names(project2)
+    clear_samples(ClustererNamespace.TRANSACTIONS, project2)
+
+
+def test_clear_redis_projects():
+    project1 = Project(id=101, name="p1", organization=Organization(pk=66))
+    project2 = Project(id=102, name="p2", organization=Organization(pk=66))
+    client = get_redis_client()
+
+    assert not client.exists(_get_projects_key(ClustererNamespace.TRANSACTIONS))
+    assert not client.exists(_get_redis_key(ClustererNamespace.TRANSACTIONS, project1))
+    assert not client.exists(_get_redis_key(ClustererNamespace.TRANSACTIONS, project2))
+
+    _record_sample(ClustererNamespace.TRANSACTIONS, project1, "foo")
+
+    assert client.exists(_get_projects_key(ClustererNamespace.TRANSACTIONS))
+    assert client.exists(_get_redis_key(ClustererNamespace.TRANSACTIONS, project1))
+    assert not client.exists(_get_redis_key(ClustererNamespace.TRANSACTIONS, project2))
+
+    _record_sample(ClustererNamespace.TRANSACTIONS, project2, "bar")
+
+    assert client.exists(_get_projects_key(ClustererNamespace.TRANSACTIONS))
```

Comment:
Should we assert the return value of `_get_projects_key` to verify both projects are there?